### PR TITLE
fix: fetch all repos for top languages

### LINKED ISSUE #1177: The most used languages only count the oldest 100 projects
**Describe the bug**

The most used languages count only count the oldest 100 projects sorted by time. The new projects are not counted

**Expected behavior**

The most used languages count should count all projects

**Screenshots / Live demo link (paste the github-readme-stats link as markdown image)**

I test it in https://docs.github.com/en/graphql/overview/explorer

The current graphql is like

```graphql
{
  user(login: "tc-imba") {
    repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

github limit the `first` to be 100, but we can add a parameter `start` to use the pagination and traverse all pages.

For example, we can add edges { cursor } and get a list of cursors

```graphql
{
  user(login: "tc-imba") {
    repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
      edges {
        cursor
      }
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

It shows that the cursor of the last repo is `Y3Vyc29yOnYyOpHOCWwxJg==`, and then we can use `start: "Y3Vyc29yOnYyOpHOCWwxJg=="` to find the 101-200 repos, and so on

**Additional context**
I'll draft a PR for this later

### PR DESCRIPTION
**Pull request by tc-imba**
This fixed #1177 

Before:
[Top Langs](https://github.com/anuraghazra/github-readme-stats)

After:
[Top Langs](https://github.com/anuraghazra/github-readme-stats)

PS. I have some python repos recently but not shown in the first figure because they are not in the oldest 100 repos. In the second figure they are successfully found.

### CODE REVIEW

**Comment by tc-imba** [comment]
@anuraghazra any comments?

**Comment by IamManchanda** [comment]
any updates to this?

**Comment by Jordhan-Carvalho** [comment]
My status is off by a considerable margin because of this, please give it some attention since it`s a critical bug for the cards

**Comment by rickstaa** [comment]
@Jordhan-Carvalho Thanks for your comment. We sadly did not have the time to look at this pull request, but we are aware of the problem (see #1852 and #1801). To get this feature implemented quicker, people can best show their support by giving a 👍🏻 to #1178 and #1801. This way, it is pushed higher in the [Top features dashboard](https://github.com/anuraghazra/github-readme-stats/issues/1935).

File: src/fetchers/top-languages-fetcher.js [hunk 1]
```diff
@@ -37,14 +43,28 @@ const fetcher = (variables, token) => {
 async function fetchTopLanguages(username, exclude_repo = []) {
   if (!username) throw Error("Invalid username");
 
-  const res = await retryer(fetcher, { login: username });
+  const pageSize = 100;
```
Comment (rickstaa) [inline src/fetchers/top-languages-fetcher.js:46]: Maybe we can use the [pageInfo](https://docs.github.com/en/graphql/reference/objects#pageinfo) object to get cleaner code. 🚀 

```js
  // Loop through all pages and fetch all repos
  let repoNodes = [];
  let hasNextPage = true
  let endCursor = null
  while (hasNextPage) {
      const variables = { login: username, first: 100, after: endCursor };
      const res = await retryer(fetcher, variables);
      
      if (res.data.errors) {
        logger.error(res.data.errors);
        throw Error(res.data.errors[0].message || "Could not fetch user");
      }
      
      repoNodes.push(...res.data.data.user.repositories.nodes);
      hasNextPage = res.data.data.user.repositories.pageInfo.hasNextPage
      endCursor = res.data.data.user.repositories.pageInfo.endCursor
  }
```

**Review by rickstaa — CHANGES_REQUESTED** [review]

**Comment by rickstaa** [comment]
@tc-imba, @Jordhan-Carvalho, I found some time today to review this pull request. Overall it looks good to me! 🚀 I would, however, use the [pageInfo](https://docs.github.com/en/graphql/reference/objects#pageinfo) object to get cleaner code. @tc-imba is there a reason you are using the cursor instead?

**Comment by rickstaa** [comment]
replaced by #2111.

**Pull request closed without merging**